### PR TITLE
feat: add version tagging

### PR DESCRIPTION
**Pull request by alaincormier-ssc**
add version-tagging sample pipelines and associated script

add release-please GitHub workflow to manage the releases of this repo.

### CODE REVIEW

**Review by lucstjean-ssc — APPROVED** [review]

**Review by davelanglois-ssc — DISMISSED** [review]

**Review by ovidiuasiminei-ssc — APPROVED** [review]

File: pipeline-samples/version-tagging/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Version Tagging
+A sample for a custom (basic) way to manage git tags and a change log.
+
+It follows [semantic versioning](https://semver.org/) (MAJOR.MINOR.PATCH).
```
Comment (Michael-McAuley-SSC) [inline pipeline-samples/version-tagging/README.md:4]: Consider replacing "it" with whatever "it" is.

**Review by Michael-McAuley-SSC — COMMENTED** [review]
Good job. Just a couple of consider type comments.

File: pipeline-samples/version-tagging/.azure-pipelines/version-tagging.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# can be set to false to disable the check for an entry in CHANGELOG.md
+variables:
+  VALIDATE_CHANGELOG: 'true'
+
+# a "Build Validation Policy" (PR trigger) must also be created
+trigger:
+- main
+
+pool:
+  vmImage: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (alaincormier-ssc) [reply pipeline-samples/version-tagging/.azure-pipelines/version-tagging.yaml]: thanks, good point
updated to `ubuntu-22.04`

please review and resolve comment if OK.
thanks

File: pipeline-samples/version-tagging/.github/workflows/version-tagging.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  # This workflow contains a single job called "version_tagging" and needs write permission
+  version_tagging:
+    permissions:
+      contents: write
+
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (alaincormier-ssc) [reply pipeline-samples/version-tagging/.github/workflows/version-tagging.yaml]: thanks, good point
updated to `ubuntu-22.04`

please review and resolve comment if OK.
thanks

File: pipeline-samples/version-tagging/README.md (hunk 1 above)
Comment (alaincormier-ssc) [reply pipeline-samples/version-tagging/README.md]: replaced "it" with "the pipeline"
please review and resolve comment if OK.
thanks

**Review by Michael-McAuley-SSC — APPROVED** [review]
Approved, thanks for the changes to versions.

**Review by davelanglois-ssc — APPROVED** [review]

**Pull request merged**